src/transformers/modeling_tf_utils.py

```diff
@@ -423,13 +424,13 @@ def input_processing(func, config, input_ids, **kwargs):
         )
         output["past_key_values"] = kwargs["kwargs_call"].pop("decoder_cached_states")
 
-    if "past" in kwargs["kwargs_call"] and "past_key_values" in kwargs:
+    if "past" in kwargs["kwargs_call"] and "past_key_values" in parameter_names:
```

Comment:
This was the root cause for the problem in the decorator -- previously, this function was called inside `call`, where `kwargs` contained arguments with their default value. 

The decorator now calls this before `call` and, because it does not have default values, `kwargs` was empty. This meant that the `past`<>`past_key_values` magic, needed for gpt2+encoder_decoder, was not happening when the decorator was applied on gpt2.

Reply:
Makes sense!